03-Bash/kernel_git/kernel_stats.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# $1 - linux kernel sources path
+
+if [ $# -lt 1 ]
+then
+    echo "set kernel path as parametr"
+    exit 1
+fi
+
+echo "*.c :$(find "$1" -name "*.c" -type f | wc -l)"
+echo "*.cpp :$(find "$1" -name "*.cpp" -type f | wc -l)"
+echo "*.c :$(find "$1" -name "*.py" -type f | wc -l)"
+
+cd "$1"
+git checkout master 2&>1 /dev/null
+
+echo "revert commit count: $(git log --oneline | grep revert | wc -l)"
+
+git ls-files '*.py' | while read py_file
+do
+    echo "in file $py_file this autors :"
+    echo "$(git blame --line-porcelain $py_file | grep '^author ' | sed 's/^author //' | sort -f | uniq -c | sort -nr)"
```

Comment:
This script gets statistic for each file separately. But statistic for all files together is required for this task.

Reply:
Done. Separated statistics collected in temporary file. Then uniq autors name counted by 'uniq'.